### PR TITLE
Fixes some Alternate Appearance issues

### LINKED ISSUE #16860: Cardborg alternate appearance can sometimes remain after removing the suit/helmet.
It happened to one person just this round. Reconnecting fixed the issue, but the player couldn't reproduce the bug again. I also couldn't reproduce the bug, but after a while, I got back to my mob with silicon appearance in the thunderdome, removed the helmet and it worked just once for me... 

@RemieRichards

### LINKED ISSUE #18941: Cardborg Disguise bug
Upon wearing the full cardborg disguise and then going into a locker, removing parts of the disguise while inside the locker and then opening it will cause you to still be stuck looking like a cyborg, changing clothing or reentering the locker will not remove the disguise.

edit: this only applies to the client, other people see it as normal

### PR DESCRIPTION
**Pull request by RemieRichards**
:cl:
fix: fixed Kor's plant disguises
fix: fixed some issues with Alternate Appearances not being added/removed
/:cl:
- Fixes Viewed Alternate Appearances not being cleared out during an atom's Destroy()
- Fixes @KorPhaeron's AA plants
- Fixes #18941 
- Fixes #16860

miss master of disguise

Thanks to our friends at paradise for helping me discover some of these issues.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
Thanks

File: code/game/atoms.dm [hunk 1]
```diff
@@ -27,7 +27,10 @@
 			var/datum/alternate_appearance/AA = alternate_appearances[aakey]
 			qdel(AA)
 		alternate_appearances = null
-
+	if(viewing_alternate_appearances)
+		for(var/aakey in viewing_alternate_appearances)
+			for(var/aa in viewing_alternate_appearances[aakey])
+				qdel(aa)
```
Comment (Krausus) [inline code/game/atoms.dm]: Won't this delete an alternate appearance for *all* viewers if any one of the viewers gets deleted?

Comment (RemieRichards) [inline code/game/atoms.dm]: *sigh* yes, it was late, ok?

**Comment by optimumtact** [comment]
Thank mr Skeletal

**Pull request merged**